### PR TITLE
blas/netlib: update cblas.h and adjust blas.go generation

### LINKED ISSUE #86: netlib/blas: adapting to upstream change of CBLAS_INDEX
In https://github.com/Reference-LAPACK/lapack/pull/588 the reference CBLAS changed the definition of CBLAS_INDEX from int to size_t. Should we update cblas.h to the latest upstream version and so adopt this change?

### PR DESCRIPTION
**Pull request by vladimir-ch**
Closes #86 

Please take a look.

### CODE REVIEW

File: blas/netlib/cblas.h [hunk 1]
```diff
@@ -1,71 +1,91 @@
...
+typedef enum CBLAS_LAYOUT {CblasRowMajor=101, CblasColMajor=102} CBLAS_LAYOUT;
+typedef enum CBLAS_TRANSPOSE {CblasNoTrans=111, CblasTrans=112, CblasConjTrans=113} CBLAS_TRANSPOSE;
+typedef enum CBLAS_UPLO {CblasUpper=121, CblasLower=122} CBLAS_UPLO;
+typedef enum CBLAS_DIAG {CblasNonUnit=131, CblasUnit=132} CBLAS_DIAG;
+typedef enum CBLAS_SIDE {CblasLeft=141, CblasRight=142} CBLAS_SIDE;
 
-int cblas_errprn(int ierr, int info, char *form, ...);
+#define CBLAS_ORDER CBLAS_LAYOUT /* this for backward compatibility with CBLAS_ORDER */
+
+/* #include "cblas_mangling.h" */
```
Comment (vladimir-ch) [inline blas/netlib/cblas.h:32]: This is the only manual edit, should I document it somehow?

Comment (kortschak) [reply blas/netlib/cblas.h:32]: It would be good to know why it's the way it is.

**Comment by kortschak** [comment]
Can you note the mechanical translations you did in the commit message?

**Comment by kortschak** [comment]
Test failures look normal.

File: blas/netlib/cblas.h (hunk 1 above)
Comment (vladimir-ch) [reply blas/netlib/cblas.h:34]: I've added a comment, PTAL.

**Comment by vladimir-ch** [comment]
> Can you note the mechanical translations you did in the commit message?

Done.

Interestingly, I noticed that the CBLAS standard specifies that I?AMAX routines return 0 for zero-length vectors (and not an invalid index like Fortran or our Go implementation) because, of course, size_t.

PTAL

**Review by kortschak — APPROVED** [review]

**Pull request merged**